session.go

```diff
@@ -299,6 +300,14 @@ func (s *Session) SetTrace(trace Tracer) {
 	s.mu.Unlock()
 }
 
+// SetQueryObserver sets the default query-level observer for this session. This setting can also
+// be changed on a per-query basis.
+func (s *Session) SetQueryObserver(observer QueryObserver) {
```

Comment:
I eventually want to remove the option to set globals on the session to remove the lock on querying #357, please remove this part. Then the observer only need to live on the config.